src/itsdangerous/serializer.py

```diff
@@ -10,13 +10,20 @@
 from .signer import _make_keys_list
 from .signer import Signer
 
+_TAnyStr = t.TypeVar("_TAnyStr", str, bytes, covariant=True)
```

Comment:
```suggestion
```
I would get rid of this, just use `t.AnyStr`, this actually needs to be invariant. Alternatively you could replace it with a `typing_extensions.TypeVar` and set `default=str` instead of `covariant=True`, that may lead to more robust results compared to relying on just the overloads on `__init__`.

Reply:
I got errors from mypy that it had to be covariant.

Reply:
That was because you weren't using it for the `payload` argument.

Reply:
After testing again I'm not getting a covariance error from mypy anymore when using `AnyStr`, not sure what was going on before.

Reply:
For `Protocol` mypy does a variance calculation and gives an error if the variance doesn't match. Since you previously only used `_TAnyStr` in return type annotations it was calculated as covariant, as soon as you added an argument that uses the same `TypeVar` the variance calculation changed to invariant.

That's why it's sometimes dangerous to re-use the same `TypeVar` between a `Protocol` and a `Generic`, since the variance may be wrong for the `Generic`, but mypy does not do a variance calculation for `Generic`. PEP-695 should make this less of a problem in the future, since type vars will generally be scoped to the generic itself, rather than in global scope and all they use auto-variance by default, so you don't have to think about it.